execute/plugin.go

```diff
@@ -205,32 +213,57 @@ func getPendingExecutedReports(
 
 		ranges, err := computeRanges(reports)
 		if err != nil {
-			return nil, nil, fmt.Errorf("compute report ranges: %w", err)
+			return nil, nil, nil, fmt.Errorf("compute report ranges: %w", err)
 		}
 
-		executedMessageSet := mapset.NewSet[cciptypes.SeqNum]()
+		// Get both finalized and unfinalized executed messages
+		finalizedMsgSet := mapset.NewSet[cciptypes.SeqNum]()
+		allExecutedMsgSet := mapset.NewSet[cciptypes.SeqNum]()
+
 		for _, seqRange := range ranges {
```

Comment:
Maybe not a problem for 6 chains, but we are sequentially looping over IO operations and that might slow the plugin down

Reply:
On the second thought, could that be replaced with 2 queries instead of 2 * len(ranges)?
Sounds like something that could bet done like this, no?

```go
allMessages, err := ccipReader.ExecutedMessages(ctx, selector, range1 or range2 or .... or rangeN, uncorfirmed)

finalized, err := ccipReader.ExecutedMessages(ctx, selector, range1 or range2 or .... or rangeN, finalized)
```

If that makes sense, can we have a ticket for that? I believe it should not be part of this PR. Of course chain reader, needs to support that types of queries